app/controllers/admin/change_reports_controller.rb

```diff
@@ -17,5 +17,24 @@ class ChangeReportsController < Admin::ApplicationController
 
     # See https://administrate-prototype.herokuapp.com/customizing_controller_actions
     # for more information
+
+    def show
+      respond_to do |format|
+        format.html do
+          super
+        end
+        format.pdf do
+          change_report = ChangeReport.find(params[:id])
+          pdf_from_html = render_to_string(
+            pdf: "change_report",
+            template: "admin/change_reports/show",
+            locals: { change_report: ChangeReportDecorator.new(change_report) },
+          )
+
+          pdf = PdfBuilder.new(pdf_from_html: pdf_from_html, attachments: change_report.pdf_letters).run
```

Comment:
I'm reading this as the attachments only being the `pdf_letters` (document type `application/pdf`). What happens to the documents uploaded as JPEG files?

Reply:
@luigi the images are rendered into the show.pdf.erb and so they get included in the pdf_from_html. I thought that was the simplest thing to do, but open to changing it if it's confusing?

Reply:
I think that's a perfectly good way to do it! I just missed it when reading through the ERB file. :+1: